### PR TITLE
Convert tests from Minitest to RSpec

### PR DESCRIPTION
**Pull request by stewart**
Mostly the same with a few syntax differences between Minitest and RSpec.

Times:

Minitest:

```
bundle exec rake test  1.37s user 0.21s system 112% cpu 1.406 total
```

RSpec:

```
rspec  0.57s user 0.10s system 109% cpu 0.609 total
```

### CODE REVIEW

File: gitnesse.gemspec [hunk 1]
```diff
@@ -20,6 +20,7 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency("minitest-matchers")
```
Comment (deadprogram) [inline gitnesse.gemspec:20]: You should be able to remove this by switching to rspec?

File: gitnesse.gemspec [hunk 2]
```diff
@@ -20,6 +20,7 @@ Gem::Specification.new do |gem|
...
   gem.add_development_dependency("mocha")
```
Comment (deadprogram) [inline gitnesse.gemspec:21]: This too?

**Comment by deadprogram** [comment]
I'm a little bit conflicted on this one. On the one hand, I understand that we can gain some benefits. Also, I like the way the specs read. I did expect faster performance from running minispec vs. rspec.

On the other, are we dragging in additional dependencies here? Is anyone actually using cucumber without also using rspec? Does that even matter really, since this is a development dependency anyhow?

**Comment by stewart** [comment]
I'm going to close this, I don't think RSpec is really needed for now, and it can always be added in later if it makes sense then.

**Pull request closed without merging**